File: 14_Longest_Common_Prefix.java
/*
Horizontal Scanning (with recursion)
*/
//Since there is no slice in Java, it's not efficient. I will give it in JavaScript.

/*
Vertical Scanning
*/

class Solution {
    public String longestCommonPrefix(String[] strs) {
        if (strs.length==0) {
            return "";
        }
        else if (strs.length==1){
            return strs[0];
        }
        String pref = "";
        int minLen = strs[0].length();
        for (int i=1; i<strs.length; i++){
            minLen = Math.min(minLen, strs[i].length());
        }
        for (int j=0; j<minLen; j++){
            char c = strs[0].charAt(j);
            for (int i=1; i<strs.length; i++){
                if (strs[i].charAt(j)!=c){
                    return pref;
                }
            }
            pref += c;
        }
        return pref;
    }
}

File: 14_Longest_Common_Prefix.js
/*
Horizontal Scanning (with recursion)
*/

/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function(strs) {
        if (strs.length==0) {
            return "";
        }
        else if (strs.length==1){
            return strs[0];
        }
        var pref = "";
        for (var i=0; i<Math.min(strs[0].length,strs[1].length); i++){
            if (strs[0].charAt(i)==strs[1].charAt(i)){
                pref += strs[0].charAt(i);
            }else{
                break;
            }
        }
        strs[1] = pref;
        return longestCommonPrefix(strs.slice(1));
};